tests/Aspire.Hosting.Tests/Cosmos/CosmosFunctionalTests.cs

```diff
@@ -26,15 +24,9 @@ public async Task VerifyCosmosWorks()
 
         using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(6));
 
-        await RetryPolicy.Handle<HttpRequestException>()
```

Comment:
Just curious - why remove this? Is there a high cost to having this when the service starts up fast enough?

Reply:
> Retrying on any HttpRequestException means when this test is actually broken the test will retry and retry over and over until the timeout expires.

That makes sense.

> The retry/waiting for the Cosmos emulator is already happening in the Cosmos client. It will retry until the server comes up.

Ok, for other cases like mysql, and for cosmos (real) the user would be responsible for adding such retries? Do they have a way of differentiating and retrying only for "connection errors", and not *any* `HttpRequestException`?

Reply:
> for other cases like mysql, and for cosmos (real) the user would be responsible for adding such retries?

it all depends on the client you are using. StackExchange.Redis has an option for it to do the retries, so we enable that option by default:
https://github.com/dotnet/aspire/blob/5043b31f5e4a50cf489ccb60c129d3edb1b42c0b/src/Components/Aspire.StackExchange.Redis/AspireRedisExtensions.cs#L201-L202

RabbitMQ doesn't, so we have to write it ourselves:

https://github.com/dotnet/aspire/blob/5043b31f5e4a50cf489ccb60c129d3edb1b42c0b/src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs#L154-L180

> Do they have a way of differentiating and retrying only for "connection errors", and not any HttpRequestException?

Yes, Polly has the feature that you can supply a predicate and inspect the Exception to check for what kind of failure - and only retry on certain exceptions.

Reply:
Thank you for the responses!